ngraph/core/src/function.cpp

```diff
@@ -405,4 +405,18 @@ void Function::remove_result(const std::shared_ptr<op::Result>& result)
         m_results.end());
 }
 
+void Function::add_parameters(const ParameterVector& params)
+{
+    m_parameters.insert(m_parameters.end(), params.begin(), params.end());
+}
+
+void Function::remove_parameter(const std::shared_ptr<op::Parameter>& param)
+{
+    m_parameters.erase(
+        std::remove_if(m_parameters.begin(),
+                       m_parameters.end(),
+                       [&param](std::shared_ptr<op::v0::Parameter>& r) { return r == param; }),
```

Comment:
`r == param` comparison will compare pointers but not Nodes if I remember right. Please doublecheck.

Reply:
right, pointers are compared here. Should (and could) we compare nodes here? In similar places (for example, validate_nodes_and_infer_types) comparing of pointers used too. Have not seen comparing of nodes itself in function.cpp.

Reply:
Yes you are right, in this case we remove node if it is a copy of exsisting shared_ptr.